assets/src/edit-story/elements/image/edit.js

```diff
@@ -41,25 +48,172 @@ const FadedImg = styled.img`
 	${ ImageWithScale }
 `;
 
-const ActualImg = styled.img`
+const CropImg = styled.img`
 	position: absolute;
 	${ ImageWithScale }
 `;
 
-function ImageEdit( { src, origRatio, width, height, x, y, scale, focalX, focalY, rotationAngle, isFullbleed } ) {
+function ImageEdit( { id, src, origRatio, width, height, x, y, scale, focalX, focalY, rotationAngle, isFullbleed } ) {
+	const [ fullImage, setFullImage ] = useState( null );
+	const [ croppedImage, setCroppedImage ] = useState( null );
+	const [ cropBox, setCropBox ] = useState( null );
+
+	const { actions: { setPropertiesById } } = useStory();
+
 	const elementProps = getBox( { x, y, width, height, rotationAngle, isFullbleed } );
 	const imgProps = getImgProps( elementProps.width, elementProps.height, scale, focalX, focalY, origRatio );
+
+	const resizeMoveableRef = useRef();
+	const panMoveableRef = useRef();
+	const translateRef = useRef( [ 0, 0 ] );
+	const cropRef = useRef( [ 0, 0, 0, 0 ] );
+
+	const updatePan = () => {
+		const [ tx, ty ] = translateRef.current;
+		fullImage.style.transform = `translate(${ tx }px, ${ ty }px)`;
+		croppedImage.style.transform = `translate(${ tx }px, ${ ty }px)`;
+	};
+
+	// Refresh moveables to ensure that the selection rect is always correct.
+	useEffect( () => {
+		if ( resizeMoveableRef.current ) {
+			resizeMoveableRef.current.updateRect();
+		}
+		if ( panMoveableRef.current ) {
+			panMoveableRef.current.updateRect();
+		}
+	} );
+
 	return (
 		<Element { ...elementProps }>
-			<FadedImg src={ src } { ...imgProps } />
-			<ActualBox>
-				<ActualImg src={ src } { ...imgProps } />
-			</ActualBox>
+			<FadedImg ref={ setFullImage } draggable={ false } src={ src } { ...imgProps } />
+			<CropBox ref={ setCropBox }>
+				<CropImg ref={ setCroppedImage } draggable={ false } src={ src } { ...imgProps } />
+			</CropBox>
+
+			{ /* Resizable moveable for cropping */ }
+			{ ! isFullbleed && (
+				<Movable
```

Comment:
Commented elsewhere as well but perhaps we could have all the Movable implementations under the one `Movable` component (a separate file per each implementation)? Would make these reusable (where applicable), and also keep the components smaller for testing/maintenance purposes.

At least the `Edit` for image is growing quite big like this.

Thoughts?

Reply:
It could be possible as long as we use separate `Moveable` components behind the scenes. But one argument here: the implementation seems fairly closely connected with the very specific structure of the `Edit` component: the moveables manipulate separate targets in the subtree. We could also move these movables into separate modules, e.g. `EditResizeMovable` and `EditPanMovable`?

Reply:
> It could be possible as long as we use separate Moveable components behind the scenes.

Yes, I was thinking exactly that -- having separate Moveable components behind the scene.

>  But one argument here: the implementation seems fairly closely connected with the very specific structure of the Edit component: the moveables manipulate separate targets in the subtree

Could the Video element or some of the shapes potentially have a similar editing experience?

> We could also move these movables into separate modules, e.g. EditResizeMovable and EditPanMovable?

Could also be an option. These could potentially then just live together in one folder instead of using one `Moveable` with separate components behind the scenes.

Reply:
@miina I moved the movables into separate modules. PTAL.

Reply:
LGTM!